x/gov/keeper/deposit.go

```diff
@@ -162,6 +163,76 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// chargeDeposit will charge proposal cancellation fee (deposits * proposal_cancel_burn_rate)  and
+// send to a destAddress if defined or burn otherwise.
+// Remaining funds are send back to the depositor.
+func (keeper Keeper) chargeDeposit(ctx sdk.Context, proposalID uint64, destAddress, proposalCancelRate string, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	rate := sdk.MustNewDecFromStr(proposalCancelRate)
+	var cancellationCharges sdk.Coins
+
+	if rate.IsPositive() {
+		for _, deposit := range totalDeposits {
+			burnAmount := sdk.NewCoin(
+				deposit.Denom,
+				sdk.NewDecFromInt(deposit.Amount).Mul(rate).RoundInt(),
+			)
+			cancellationCharges = cancellationCharges.Add(burnAmount)
+		}
+	}
+
+	// burn the cancellation fee or sent the cancellation charges to destination address.
+	if !cancellationCharges.IsZero() {
+		// get the distribution module account address
+		distributionAddress := keeper.authKeeper.GetModuleAddress(disttypes.ModuleName)
+		if distributionAddress.String() == destAddress {
+			err := keeper.distrkeeper.FundCommunityPool(ctx, cancellationCharges, keeper.ModuleAccountAddress())
+			if err != nil {
+				return err
+			}
+		} else {
+			if len(destAddress) == 0 {
```

Comment:
```suggestion
		} else if len(destAddress) == 0 {
```

Propose to use `else if` to keep indentations level low. Could also consider `switch` as @atheeshp proposed.